src/urql/atomWithSubscription.ts

```diff
@@ -0,0 +1,69 @@
+import { pipe, subscribe } from 'wonka'
+import {
+  TypedDocumentNode,
+  OperationContext,
+  OperationResult,
+} from '@urql/core'
+import { atom } from 'jotai'
+import type { Atom, Getter } from 'jotai'
+import { getClient } from './clientAtom'
+
+type SubscriptionArgs<Data, Variables extends object> = {
+  query: TypedDocumentNode<Data, Variables>
+  variables?: Variables
+  context?: Partial<OperationContext>
+}
+
+export function atomWithSubscription<Data, Variables extends object>(
+  createSubscriptionArgs: (get: Getter) => SubscriptionArgs<Data, Variables>
+): Atom<OperationResult<Data, Variables>> {
+  const operationResultAtom = atom<
+    OperationResult<Data, Variables> | Promise<OperationResult<Data, Variables>>
+  >(
+    new Promise<OperationResult<Data, Variables>>(() => {}) // infinite pending
+  )
+  const queryAtom = atom((get) => {
+    const args = createSubscriptionArgs(get)
+    const initAtom = atom(
+      null,
+      (get, set, cleanup: (callback: () => void) => void) => {
+        set(
+          operationResultAtom,
+          new Promise<OperationResult<Data, Variables>>(() => {}) // new fetch
+        )
+        const client = getClient(get, set)
+        const subscription = pipe(
+          client.subscription(args.query, args.variables, args.context),
```

Comment:
atomWithSubscription is almost like atomWithQuery without `requestPolicy` in args.